third-party/realdds/src/dds-network-adapter-watcher.cpp

```diff
@@ -72,8 +72,15 @@ class network_adapter_watcher_singleton
     ~network_adapter_watcher_singleton()
     {
         _adapter_watcher.reset();  // signal the thread to finish
-        if( _th.joinable() )
-            _th.join();
+        try 
+        {
+            if (_th.joinable())
+                _th.join();
+        }
+        catch (std::system_error& e)
+        {
+            LOG_ERROR("join failed, thread might no longer be joinable, Error:" << e.what());
```

Comment:
Not sure this will work.
Can you try to see if it works and you see the e.what()?
You might need to use `rsutils::from() << `

Reply:
LOG_ERROR has `<<`, if that was the concern

Reply:
I would ask that we make error messages a little nicer, though: capitalization, e.g.:
`LOG_ERROR( "Network adapter watch termination failed: " << e.what() );`
@Nir-Az up to you whether this is an error the user should see - we might want to leave it debug?

Reply:
Oh OK

Reply:
Sure why not, @AviaAv  can you handle please?